nav2_behavior_tree/plugins/action/back_up_action.cpp

```diff
@@ -51,6 +51,14 @@ class BackUpAction : public BtActionNode<nav2_msgs::action::BackUp>
     goal_.speed = speed;
   }
 
+  void on_tick() override
+  {
+    int recovery_count = 0;
```

Comment:
All this recovery count stuff would be good to push to a base class or helper function so it doesn't need to be repeated.

Reply:
Good point